core/src/main/scala/feral/IOSetup.scala

```diff
@@ -20,24 +20,28 @@ import cats.effect.unsafe.IORuntime
 import cats.effect.kernel.Resource
 import cats.effect.IO
 import cats.effect.kernel.Deferred
+import cats.effect.std.Dispatcher
 
 private[feral] trait IOSetup {
+  private def runtime =
+    IORuntime.global
 
-  protected def runtime: IORuntime = IORuntime.global
+  protected def dispatcher: Dispatcher[IO] =
+    Dispatcher.sequential[IO](await = true).allocated.map(_._1).unsafeRunSync()(runtime)
```

Comment:
`await = true` doesn't really matter, since we are discarding the finalizer anyway.

Instead of `sequential` I would lean for `parallel`. It slightly increases the overhead per invocation, but it avoids the situation where one task prevents another task from running.

Also we can't use `unsafeRunSync()` anywhere in this code, because it must cross-compile for JS as well.